internal/telemetry/telemetry.go

```diff
@@ -0,0 +1,229 @@
+				body, err := json.Marshal(event)
+				if err != nil {
+					t.logger.Debug("failed to marshal event", zap.Error(err))
+					continue
+				}
+
+				req, err := http.NewRequestWithContext(ctx, "POST", t.config.HookdeckSourceURL, bytes.NewBuffer(body))
+				if err != nil {
+					t.logger.Debug("failed to create request", zap.Error(err))
+					continue
+				}
+				req.Header.Set("Content-Type", "application/json")
+
+				resp, err := t.client.Do(req)
+				if err != nil {
+					t.logger.Debug("failed to send event", zap.Error(err))
+					continue
+				}
+				if resp.StatusCode >= 400 {
+					t.logger.Debug("failed to send event",
+						zap.Int("status", resp.StatusCode),
+						zap.Any("event", event))
+				}
+				resp.Body.Close()
+			}
+		}()
+	}
+
+	// Send jobs to workers
+	for _, event := range batch {
+		jobs <- event
+	}
+	close(jobs)
+
+	// Wait for all workers
+	wg.Wait()
+}
+
+func (t *telemetryImpl) Init(ctx context.Context) {
+	if err := sentry.Init(sentry.ClientOptions{
+		Dsn:              t.config.SentryDSN,
+		EnableTracing:    true,
+		TracesSampleRate: 1, // FIXME
+		AttachStacktrace: true,
+	}); err != nil {
+		t.logger.Error("sentry initialization failed", zap.Error(err))
+	}
+
+	t.eventChan = make(chan telemetryEvent, t.config.BatchSize*10)
+	t.done = make(chan struct{})
+	t.client = &http.Client{
+		Timeout: 10 * time.Second,
+		Transport: &http.Transport{
+			MaxIdleConns:        100,
+			MaxIdleConnsPerHost: 100,
+			IdleConnTimeout:     90 * time.Second,
+		},
+	}
+	go t.processEvents()
+}
+
+func (t *telemetryImpl) Flush() {
+	close(t.done)
+	sentry.Flush(2 * time.Second)
+}
+
+func (t *telemetryImpl) MakeSentryHandler() gin.HandlerFunc {
+	return sentrygin.New(sentrygin.Options{
+		Repanic: true,
+	})
+}
+
+func (t *telemetryImpl) ApplicationStarted(ctx context.Context, application ApplicationInfo) {
+	event := telemetryEvent{
+		EventType: "application_started",
+		Payload:   application.ToPayload(),
+		Time:      time.Now(),
+	}
+	t.sendEvent(event)
+}
+
+func (t *telemetryImpl) DestinationCreated(ctx context.Context, destinationType string) {
+	event := telemetryEvent{
+		EventType: "destination_created",
+		Payload:   map[string]interface{}{"type": destinationType},
+		Time:      time.Now(),
+	}
+	t.sendEvent(event)
+}
+
+func (t *telemetryImpl) TenantCreated(ctx context.Context) {
+	event := telemetryEvent{
+		EventType: "tenant_created",
+		Payload:   map[string]interface{}{},
+		Time:      time.Now(),
+	}
+	t.sendEvent(event)
+}
+
+type ApplicationInfo struct {
+	Version       string
+	MQ            string
+	PortalEnabled string
+}
+
+func (a *ApplicationInfo) ToPayload() map[string]interface{} {
+	return map[string]interface{}{
+		"version":        a.Version,
+		"mq":             a.MQ,
+		"portal_enabled": a.PortalEnabled,
+	}
+}
+
+type telemetryEvent struct {
+	EventType string                 `json:"type"`
+	Payload   map[string]interface{} `json:"payload"`
+	Time      time.Time              `json:"time"`
+}
```

Comment:
I think the Installation ID should be at this top level along with `type` and `payload`.

Reply:
`time` -> `timestamp`. Assuming this is an ISO 8601 timestamp?

Reply:
nit: I feel like `payload` should just be `data`.

Reply:
Updated in the latest version, can you go over them and resolve this if the change looks good?